ubuntudesign/developer.ubuntu.com
# Serve static files in production

Use WhiteNoise to serve static files in production.

QA
==

Check that static files are served correctly and DEBUG mode is set correctly.

Local development
--- 

Run the local development server:

``` bash
make run
```

Now:

- Visit http://127.0.0.1:8017/static/developer_portal/css/devportal.css and see that it is there.
- Check debug mode is *on* by visiting http://127.0.0.1:8017/en/iamnothere/ and seeing that you get a Django error page.

production
---

Run the site with gunicorn (like in production):

``` bash
make "docker-compose run --service-ports --entrypoint /bin/bash webapp"
pip install -I gunicorn
gunicorn -b 0.0.0.0:5000 developer_portal.wsgi
```

Now:

- Visit http://127.0.0.1:8017/static/developer_portal/css/devportal.css and see that it is there.
- Check debug mode is *off* by visiting http://127.0.0.1:8017/en/iamnothere/ and seeing that you get an ubuntu-styled 404 page.
